packages/docmanager/test/dialog.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { DocumentRegistry } from '@jupyterlab/docregistry';
+import { Contents, ServiceManager } from '@jupyterlab/services';
+import { Widget } from '@lumino/widgets';
+import * as Mock from '@jupyterlab/testutils/lib/mock';
+import { DocumentManager, renameFile } from '../src';
+
+function shouldOverwriteFalse(path: string): Promise<boolean> {
+  return Promise.resolve(false);
+}
+
+describe('docregistry/dialog', () => {
+  let manager: DocumentManager;
+  let services: ServiceManager.IManager;
+  let alreadyExistsError: any = {};
+
+  beforeAll(() => {
+    const registry = new DocumentRegistry({});
+    services = new Mock.ServiceManagerMock();
+    manager = new DocumentManager({
+      registry,
+      manager: services,
+      opener: {
+        open: (widget: Widget) => {
+          // no-op
+        }
+      }
+    });
+  });
+
+  beforeEach(() => {
+    alreadyExistsError = {
+      name: 'file already exists',
+      message: 'File already exists: bar.ipynb',
+      response: {
+        status: 409
+      }
+    };
+    manager.rename = (oldPath, newPath) => {
+      const renamePromise = new Promise(function (resolve, reject) {
+        throw alreadyExistsError;
+      }) as Promise<Contents.IModel>;
+      return renamePromise;
+    };
+  });
```

Comment:
You should be able to take advantage of the [jest.spyOn](https://jestjs.io/docs/26.x/jest-object#jestspyonobject-methodname) feature with a code like:

```suggestion
    alreadyExistsError = {
      name: 'file already exists',
      message: 'File already exists: bar.ipynb',
      response: {
        status: 409
      }
    };
    // spyRename can be store as variable as manager
    spyRename = jest.spyOn(manager, 'rename');
    spyRename.mockRejectedValue(alreadyExistsError);
  });

  afterEach(() => {
    spyRename.mockReset();
  });
```

Reply:
Cool, this is much simpler, thanks :). Changed in 370655d5fc.